hass_nabucasa/exceptions.py

```diff
@@ -15,6 +15,10 @@ class NabuCasaAuthenticationError(NabuCasaBaseError):
     """Base class for all Nabu Casa authentication exceptions."""
 
 
+class NabuCasaNotLoggedInError(NabuCasaBaseError):
+    """Exception raised when the user is not logged in."""
```

Comment:
We could have a default message here as this does not have any variance.
However, this is not important for this PR, but something to consider for the future.